extensions/binding/org.eclipse.smarthome.binding.weatherunderground/src/main/java/org/eclipse/smarthome/binding/weatherunderground/handler/WeatherUndergroundHandler.java

```diff
@@ -0,0 +1,325 @@
+
+            String lang = StringUtils.trimToEmpty(config.language);
+            if (lang.isEmpty()) {
+                urlStr = urlStr.replace("%SETTINGS%", "");
+            } else {
+                urlStr = urlStr.replace("%SETTINGS%", "lang:" + lang.toUpperCase());
+            }
+
+            urlStr = urlStr.replace("%QUERY%", location);
+            logger.debug("URL = {}", urlStr);
+
+            // Run the HTTP request and get the JSON response from Weather Underground
+            URL url = new URL(urlStr);
+            URLConnection connection = url.openConnection();
+            String repsonse = IOUtils.toString(connection.getInputStream());
+            logger.debug("weatherData = {}", repsonse);
+
+            // Map the JSON response to an object
+            result = gson.fromJson(repsonse, WeatherUndergroundJsonData.class);
+            if (result == null) {
+                errorMsg = "no data returned";
+            } else if (result.getResponse() == null) {
+                errorMsg = "missing response sub-object";
+            } else if (result.getResponse().getErrorDescription() != null) {
+                errorMsg = result.getResponse().getErrorDescription();
+            } else {
+                resultOk = true;
+                for (String feature : features) {
+                    if (feature.equals(FEATURE_CONDITIONS) && result.getCurrent() == null) {
+                        resultOk = false;
+                        errorMsg = "missing current_observation sub-object";
+                    } else if (feature.equals(FEATURE_FORECAST10DAY) && result.getForecast() == null) {
+                        resultOk = false;
+                        errorMsg = "missing forecast sub-object";
+                    } else if (feature.equals(FEATURE_GEOLOOKUP) && result.getLocation() == null) {
+                        resultOk = false;
+                        errorMsg = "missing forecast sub-object";
+                    }
+                }
+            }
+
+            if (!resultOk) {
+                logger.warn("Error in Weather Underground response: {}", errorMsg);
+            }
+
+        } catch (MalformedURLException e) {
+            errorMsg = e.getMessage();
+            logger.warn("Constructed url is not valid: {}", errorMsg);
+        } catch (IOException e) {
+            errorMsg = e.getMessage();
+            logger.warn("Error running Weather Underground request: {}", errorMsg);
+        } catch (JsonSyntaxException e) {
+            errorMsg = e.getMessage();
+            logger.warn("Error parsing Weather Underground response: {}", errorMsg);
+        }
+
+        // Update the thing status
+        if (resultOk) {
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, errorMsg);
+        }
+
+        return resultOk ? result : null;
```

Comment:
Why put this into a global?  You only need it when updating the channel data right?